### PR TITLE
[RHCLOUD-20691] Check err handling from Bind() in handlers

### PR DESCRIPTION
**Pull request by Ellen-Yi-Dong**
Within authentication_handlers.go, instead of util.NewErrBadRequest(err), change to err and check err handling from Bind() in handlers.

Within authentication_handlers_test.go, set binder on each invocation

JIRA: RH-CLOUD 20691[https://issues.redhat.com/browse/RHCLOUD-20691](url)

### CODE REVIEW

**Review by MikelAlejoBR — APPROVED** [review]
Nice one! Looks good to me!

File: authentication_handlers_test.go [hunk 1]
```diff
@@ -417,6 +420,8 @@ func TestAuthenticationCreateBadRequestInvalidResourceType(t *testing.T) {
 	)
 	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
 
+	c.Echo().Binder = echoBinder
+
```
Comment (petracihalova) [inline authentication_handlers_test.go:424]: well I am not sure about this ... yes we need it here but I think we could set Binder() for all our tests on one place ... I started discussion on Slack ... maybe I am wrong but this can probably lead to separate PR where we set Binder() for all our tests (so in this PR we can leave it the way you did it) 🤔 😇

File: authentication_handlers.go [hunk 2]
```diff
@@ -150,13 +150,13 @@ func AuthenticationEdit(c echo.Context) error {
...
 	}
 
-	updateRequest := &m.AuthenticationEditRequest{}
-	err = c.Bind(updateRequest)
+	updateRequest := m.AuthenticationEditRequest{}
+	err = c.Bind(&updateRequest)
 	if err != nil {
-		return util.NewErrBadRequest(err)
+		return err
 	}
```
Comment (petracihalova) [inline authentication_handlers.go:157]: we need this change in other handlers too (but can be part of another PR for sure 😇)
